specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,454 @@
+
+Finally, we require that each token ID is unique within the smart contract. This will allow us to query token only by token ID, without knowing it's kind.
+
+### SBT Registry
+
+Atomicity of _soul transfer_ in current NEAR runtime is not possible if the token balance is kept separately for each SBT smart contract. We need an additional contract: the `SBT Registry`, to provide atomic transfer of all user tokens and efficient way to block accounts in relation to a Ban event. The registry will provide a balance book for all associated SBT tokens.
+
+An SBT smart contract, SHOULD opt-in to a registry using `opt_in` function. One SBT smart contract can opt-in to:
+
+- many registries: it MUST relay all state change functions to all registries.
+- or to no registry: it MUST issue the SBT state change emits by itself.
+
+If an SBT smart contract doesn't opt-in to any registry, then we should think about it as a single token registry, and it MUST strictly implement all SBT Registry query functions by itself: the contract address must be part of the arguments, and it must check that it equals to the deployed account address (`require!(ctr == env::current_account_id())`).
+
+Moreover, a registry will provide an efficient way to query multiple tokens for a single user. This will allow implementation of use cases such us:
+
+- SBT based identities (main use case of the `i-am-human` protocol);
+- SBT classes;
+- decentralized societies.
+
+#### Recovery within an SBT Registry
+
+SBT registry can define it's own mechanism to atomically recover all tokens related to one account and execute soul transfer to another account, without going one by one through each SBT issuer (sometimes that might be even not possible). Example mechanisms a Registry can implement to recovery mechanism:
+
+- KYC based recovery
+- Social recovery
+
+![Social Recovery, Image via “Decentralized Society”](https://bankless.ghost.io/content/images/public/images/cdb1fc23-6179-44f0-9bfe-e5e5831492f7_1399x680.png)
+
+SBT Registry based recovery is not part of this specification.
+
+#### Minting authorization
+
+A registry can limit which contracts can mint SBTs by implementing a custom issuer registration methods. Example: a simple access control list managed by a DAO.
+
+### Smart contract interface
+
+For the Token ID type we propose `u64` rather than `U128`. `u64` capacity is more than 1e19. If we will mint 10'000 SBTs per second, than it will take us 58'494'241 years to get into the capacity.
+Today, the JS integer limit is `2^53-1 ~ 9e15`. It will take us 28561 years to fill that when minting 10'000 SBTs per second. So, we don't need to u128 nor a String type. However, if for some reason, we will need to get u64 support for JS, then we can always add another set of methods which will return String, so making it compatible with NFT standard (which is using `U128`, which is a string).
```

Comment:
I would consider limiting the TokenId and KindId to be explicitly within $0..2^{53}-1$ range. Then Rust implementations can safely use u64, and JS can also rely on the use of numbers, and it would be the contract fault if it uses a number that is too big.

Reply:
In 100 years the JS (if it will still exists) it will be completely different. 28'000 years is really a lot!